main.rb

```diff
@@ -0,0 +1,46 @@
+morse_code = {
+  '.-' => 'A',
+  '-...' => 'B',
+  '-.-.' => 'C',
+  '-..' => 'D',
+  '.' => 'E',
+  '..-.' => 'F',
+  '--.' => 'G',
+  '....' => 'H',
+  '..' => 'I',
+  '.---' => 'J',
+  '-.-' => 'K',
+  '.-..' => 'L',
+  '--' => 'M',
+  '-.' => 'N',
+  '---' => 'O',
+  '.--.' => 'P',
+  '--.-' => 'Q',
+  '.-.' => 'R',
+  '...' => 'S',
+  '-' => 'T',
+  '..-' => 'U',
+  '...-' => 'V',
+  '.--' => 'W',
+  '-..-' => 'X',
+  '-.--' => 'Y',
+  '--..' => 'Z'
+}
+
+def decode_char(string, morse_code)
+  print morse_code[string]
+end
+
+def decode_word(morse, morse_code)
+  print "#{morse.split.map { |char| decode_char(char, morse_code) }.join} "
+end
+
+def decode_message(message, morse_code)
+  print message.split('   ').map { |word| decode_word(word, morse_code) }.join(' ')
```

Comment:
- [ ] As the previous reviewer requested, kindly return the output of the method instead of printing it, which means your code should not have either `puts` or `print` and should look like this:

```
def decode_char(string, morse_code)
  morse_code[string]
end

def decode_word(morse, morse_code)
  morse.split.map { |char| decode_char(char, morse_code) }.join
end

def decode_message(message, morse_code)
  message.split('   ').map { |word| decode_word(word, morse_code) }.join(' ')
end